apps/prairielearn/src/lib/db-types.ts

```diff
@@ -23,13 +23,13 @@ export const IdSchema = z
  * This is a schema for the objects produced by the `postgres-interval` library.
  */
 const PostgresIntervalSchema = z.object({
-  years: z.number(),
-  months: z.number(),
-  days: z.number(),
-  hours: z.number(),
-  minutes: z.number(),
-  seconds: z.number(),
-  milliseconds: z.number(),
+  years: z.number().optional(),
+  months: z.number().optional(),
+  days: z.number().optional(),
+  hours: z.number().optional(),
+  minutes: z.number().optional(),
+  seconds: z.number().optional(),
+  milliseconds: z.number().optional(),
```

Comment:
This is now conflicting with the changes from #9066; I'd accept Jonatan's changes to have this use `.default(0)`. That will also make the below changes to `IntervlSchema` unnecessary.

Reply:
I have now pull in Jonatan's changes and reverted the changes made to `IntervalSchema`.